index-providers/modeshape-elasticsearch-index-provider/src/main/java/org/modeshape/jcr/index/elasticsearch/Operations.java

```diff
@@ -0,0 +1,351 @@
+        } 
+        
+        if (op instanceof NodePath) {
+            return PATH_BUILDER;
+        } 
+        
+        if (op instanceof NodeDepth) {
+            return DEPTH_BUILDER;
+        } 
+        
+        if (op instanceof NodeName) {
+            return NODE_NAME_BUILDER;
+        } 
+        
+        if (op instanceof LowerCase) {
+            return LOWER_CASE_BUILDER;
+        }
+        
+        if (op instanceof UpperCase) {
+            return UPPER_CASE_BUILDER;
+        }
+
+        if (op instanceof NodeLocalName) {
+            return NODE_LOCALNAME_BUILDER;
+        }
+
+        if (op instanceof Length) {
+            return LENGTH_BUILDER;
+        }
+        
+        return null;
+    }
+ 
+    private StaticOperandBuilder operand(StaticOperand op) {
+        if (op instanceof Literal) {
+            return LITERAL_BUILDER;
+        }
+        
+        if (op instanceof BindVariableValue) {
+            return BIND_VARIABLE_VALUE_BUILDER;
+        }
+        
+        return null;
+    }
+    
+    private final OperandBuilder PROPERTY_VALUE_BUILDER = new OperandBuilder<PropertyValue>() {
+        @Override
+        public String apply(PropertyValue operand, Map<String, Object> variables) {
+            return operand.getPropertyName();
+        }
+    };
+
+    private final StaticOperandBuilder LITERAL_BUILDER = new StaticOperandBuilder<Literal>() {
+        @Override
+        public Object apply(String field, Literal op, Map<String, Object> variables) {
+            EsIndexColumn col = columns.column(field);
+            
+            if (op.value() instanceof Object[]) {
+                return col.columnValue((Object[])col.cast((Object[])op.value()));
+            }
+                        
+            return col.columnValue(col.cast(op.value()));
+        }
+    };
+
+    private final StaticOperandBuilder BIND_VARIABLE_VALUE_BUILDER = new StaticOperandBuilder<BindVariableValue>() {
+        @Override
+        public Object apply(String field, BindVariableValue op, Map<String, Object> variables) {
+            EsIndexColumn col = columns.column(field);
+            Object value = variables.get(op.getBindVariableName());
+            
+            if (value instanceof StaticOperand) {
+                return operand((StaticOperand)value).apply(field, value, variables);
+            }
+            
+            if (value instanceof DynamicOperand) {
+                return operand((DynamicOperand)value).apply(value, variables);
+            }
+            
+            if (value instanceof Object[]) {
+                return col.cast((Object[]) value);
+            }
+            
+            if (value instanceof Collection) {
+                return col.cast((Collection) value);
+            }
+            
+            return col.cast(variables.get(op.getBindVariableName()));
+        }
+    };
+
+    private final OperandBuilder LOWER_CASE_BUILDER = new OperandBuilder<LowerCase>() {
```

Comment:
If these are immutable constants, they should be marked as static. If not, their name shouldn't be uppercased.